### PR TITLE
Document what ADMIN CHECK does

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Add more details to the `ADMIN CHECK TABLE` documentation.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### Notes

Some notes, if desired we could add some of this to the docs, but I don't think most of this belongs in user documentation.

In `executor/executor.go` there is `CheckTableExec` which handles `ADMIN CHECK TABLE`.

Note that it sets the concurrency to the number of indexes with a maxiumum of 3:
```go
        concurrency := mathutil.Min(3, len(e.srcs))
```

If there are no indexes it doesn't do anything.

There is some special handeling for `MVIndex` (multi-valued index).

After comparing the number of entries in the index and table it starts to compare the data with the `compareData`
method of `tableWorker` in `executor/distsql.go`.

There is a chunking mechanism (chunks of 1000 records?) that is used.

Can `tidb_index_lookup_join_concurrency` be used to speedup `ADMIN CHECK TABLE`?

Any special notes for dealing with large tables?

Running multiple `ADMIN CHECK TABLE` statements at the same time is not recommended due to resource usage (e.g. memory).

Progress reporting seems to be limited to one line per index:

```
[2023/04/27 09:38:45.026 +02:00] [INFO] [admin.go:101] ["check indices count"] [table=orders] [tblCnt=345671] [index="\"idx_order\""] [idxCnt=345671]
[2023/04/27 09:38:45.229 +02:00] [INFO] [admin.go:101] ["check indices count"] [table=orders] [tblCnt=345671] [index="\"o_all_local\""] [idxCnt=345671]
```

### CODE REVIEW

File: sql-statements/sql-statement-admin-check-table-index.md [hunk 1]
```diff
@@ -6,7 +6,24 @@ category: reference
...
 
-The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes. It does not support checking [FOREIGN KEY constraints](/foreign-key.md).
+The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes. 
+
+It does not support:
+
+- checking [FOREIGN KEY constraints](/foreign-key.md).
+- checking the PRIMARY KEY index if a [clustered primary key](/clustered-indexes.md) is used.
+
+If `ADMIN CHECK [TABLE|INDEX]` finds any issues these can be fixed by recreating the index by dropping the index and adding it again.
```
Comment (ekexium) [inline sql-statements/sql-statement-admin-check-table-index.md:16]: Or `admin recover` or `admin cleanup`.
And it is related to `troubleshoot-data-inconsistency-errors.md`. We can add a link here.

File: sql-statements/sql-statement-admin-check-table-index.md [hunk 2]
```diff
@@ -6,7 +6,24 @@ category: reference
 
 # ADMIN CHECK [TABLE|INDEX]
 
-The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes. It does not support checking [FOREIGN KEY constraints](/foreign-key.md).
+The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes.
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-admin-check-table-index.md:9]:
```suggestion
The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes.
```

File: sql-statements/sql-statement-admin-check-table-index.md [hunk 3]
```diff
@@ -6,7 +6,24 @@ category: reference
...
+The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes. 
+
+It does not support:
+
+- checking [FOREIGN KEY constraints](/foreign-key.md).
+- checking the PRIMARY KEY index if a [clustered primary key](/clustered-indexes.md) is used.
+
+If `ADMIN CHECK [TABLE|INDEX]` finds any issues these can be fixed by recreating the index by dropping the index and adding it again. 
+
+## Steps
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-admin-check-table-index.md:18]:
```suggestion
## Principles
```

File: sql-statements/sql-statement-admin-check-table-index.md (hunk 1 above)
Comment (Oreoxmt) [inline sql-statements/sql-statement-admin-check-table-index.md:16]:
```suggestion
If `ADMIN CHECK [TABLE|INDEX]` finds any issues, you can resolve them by dropping and recreating the index. 
```

File: sql-statements/sql-statement-admin-check-table-index.md [hunk 4]
```diff
@@ -6,7 +6,24 @@ category: reference
...
+
+The `ADMIN CHECK TABLE` statement takes the following steps to check the table:
+
+1. For each index it checks if the number of records in the index is the same as the number of records in the table.
+
+2. For each index it loops over the values in each row and compare the values with the values in the table.
+
+If `ADMIN CHECK INDEX` is used it only checks the specified index.
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-admin-check-table-index.md:26]:
```suggestion
1. For each index, it checks if the number of records in the index is the same as that in the table.

2. For each index, it loops over the values in each row and compares the values with that in the table.

If you use the `ADMIN CHECK INDEX` statement, it only checks the specified index.
```

File: sql-statements/sql-statement-admin-check-table-index.md [hunk 5]
```diff
@@ -6,7 +6,24 @@ category: reference
...
 
-The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes. It does not support checking [FOREIGN KEY constraints](/foreign-key.md).
+The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes.
+
+It does not support the following:
+
+- Checking [FOREIGN KEY constraints](/foreign-key.md).
+- Checking the PRIMARY KEY index if a [clustered primary key](/clustered-indexes.md) is used.
+
+If `ADMIN CHECK [TABLE|INDEX]` finds any issues, you can resolve them by dropping and recreating the index.
```
Comment (ekexium) [inline sql-statements/sql-statement-admin-check-table-index.md:16]: It is usually a TiDB bug. Should we suggest that users report the issue to PingCAP?

**Review by ekexium — APPROVED** [review]
The rest looks good.

File: sql-statements/sql-statement-admin-check-table-index.md (hunk 5 above)
Comment (Oreoxmt) [inline sql-statements/sql-statement-admin-check-table-index.md:16]:
```suggestion
If `ADMIN CHECK [TABLE|INDEX]` finds any issues, you can resolve them by dropping and recreating the index. If the issue is not resolved, you can report an [issue](https://github.com/pingcap/tidb/issues/new/choose) on GitHub.
```
@ekexium PTAL

**Review by Oreoxmt — APPROVED** [review]

File: sql-statements/sql-statement-admin-check-table-index.md [hunk 6]
```diff
@@ -6,7 +6,24 @@ category: reference
...
 
-The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes. It does not support checking [FOREIGN KEY constraints](/foreign-key.md).
+The `ADMIN CHECK [TABLE|INDEX]` statement checks for data consistency of tables and indexes.
+
+It does not support the following:
+
+- Checking [FOREIGN KEY constraints](/foreign-key.md).
+- Checking the PRIMARY KEY index if a [clustered primary key](/clustered-indexes.md) is used.
+
+If `ADMIN CHECK [TABLE|INDEX]` finds any issues, you can resolve them by dropping and recreating the index. If the issue is not resolved, you can report an [issue](https://github.com/pingcap/tidb/issues/new/choose) on GitHub.
```
Comment (ran-huang) [inline sql-statements/sql-statement-admin-check-table-index.md:16]: I'm not sure if GitHub issue is still a recommended channel for support. Maybe we should link to [support resources](https://docs.pingcap.com/tidb/stable/support) instead.

**Review by ran-huang — APPROVED** [review]
rest LGTM

File: sql-statements/sql-statement-admin-check-table-index.md (hunk 6 above)
Comment (dveeden) [reply sql-statements/sql-statement-admin-check-table-index.md:16]: Looks like this is a common sentence in SQL Statement docs:

```
$ git grep -l 'github.com/pingcap/tidb/issues/new/choose'
dr-secondary-cluster.md
sql-statements/sql-statement-admin-check-table-index.md
sql-statements/sql-statement-alter-database.md
sql-statements/sql-statement-create-database.md
sql-statements/sql-statement-create-role.md
sql-statements/sql-statement-create-table-like.md
sql-statements/sql-statement-deallocate.md
sql-statements/sql-statement-delete.md
sql-statements/sql-statement-do.md
sql-statements/sql-statement-drop-database.md
sql-statements/sql-statement-drop-role.md
sql-statements/sql-statement-drop-view.md
sql-statements/sql-statement-execute.md
sql-statements/sql-statement-flush-privileges.md
sql-statements/sql-statement-grant-role.md
sql-statements/sql-statement-insert.md
sql-statements/sql-statement-load-data.md
sql-statements/sql-statement-prepare.md
sql-statements/sql-statement-rename-index.md
sql-statements/sql-statement-rename-table.md
sql-statements/sql-statement-rename-user.md
sql-statements/sql-statement-replace.md
sql-statements/sql-statement-revoke-role.md
sql-statements/sql-statement-set-default-role.md
sql-statements/sql-statement-set-names.md
sql-statements/sql-statement-set-password.md
sql-statements/sql-statement-set-role.md
sql-statements/sql-statement-show-character-set.md
sql-statements/sql-statement-show-collation.md
sql-statements/sql-statement-show-columns-from.md
sql-statements/sql-statement-show-create-database.md
sql-statements/sql-statement-show-create-table.md
sql-statements/sql-statement-show-databases.md
sql-statements/sql-statement-show-errors.md
sql-statements/sql-statement-show-grants.md
sql-statements/sql-statement-show-indexes.md
sql-statements/sql-statement-show-plugins.md
sql-statements/sql-statement-show-privileges.md
sql-statements/sql-statement-show-table-status.md
sql-statements/sql-statement-show-tables.md
sql-statements/sql-statement-show-variables.md
sql-statements/sql-statement-show-warnings.md
sql-statements/sql-statement-shutdown.md
sql-statements/sql-statement-truncate.md
sql-statements/sql-statement-use.md
troubleshoot-tidb-cluster.md
```

Comment (dveeden) [reply sql-statements/sql-statement-admin-check-table-index.md:16]: I've created #14648 for this.

**Pull request merged**